### PR TITLE
[REVIEW] Add cugraph+UCX build instructions

### PR DESCRIPTION
**Pull request by VibhuJawa**
This PR adds the instructions to build cugraph+UCX for benchmarking . 

The current conda build does not support IB as well has challenges using NVLINK .  This PR adds the dockefile as well as build instructions and test to streamline benchmarking effort. 

The expected speedup on sampling is around `8x` by using the UCX  souce build  and a `20x` better bandwidth with compute.   

See [README](https://github.com/rapidsai/cugraph/blob/d513a86017fac6ba4742019eae47f7ffe98186e0/benchmarks/shared/build_cugraph_ucx/README.MD) for instructions . 

### `dask.compute` bandwidth
**With the standard conda build :** 
```python3
python3 test_client_bandwidth.py 

Getting 1,000,000 rows  of size 0.01 took = 168.32900047302246 ms
Bandwidth = 0.0594 gb/s
Getting 2,000,000 rows  of size 0.02 took = 260.2977991104126 ms
Bandwidth = 0.0768 gb/s
Getting 4,000,000 rows  of size 0.04 took = 288.5307312011719 ms
Bandwidth = 0.1386 gb/s
Getting 25,000,000 rows  of size 0.28 took = 1051.6365766525269 ms
Bandwidth = 0.2663 gb/s
----------------------------------------Completed Test----------------------------------------
```
**With this **PRs** Build** :
```python3
python3 test_client_bandwidth.py

Getting 1,000,000 rows  of size 0.01 took = 14.061594009399414 ms
Bandwidth = 0.7112 gb/s
Getting 2,000,000 rows  of size 0.02 took = 19.601154327392578 ms
Bandwidth = 1.0203 gb/s
Getting 4,000,000 rows  of size 0.04 took = 15.92099666595459 ms
Bandwidth = 2.5124 gb/s
Getting 25,000,000 rows  of size 0.28 took = 52.902913093566895 ms
Bandwidth = 5.2927 gb/s
----------------------------------------Completed Test----------------------------------------
```

### `cugraph sampling` benchmark

With the standard conda build : 
```python3
python3 test_cugraph_sampling.py

Sampling 1,000 took = 78.96044254302979 ms
Sampling 10,000 took = 189.63768482208252 ms
Sampling 100,000 took = 1084.9100351333618 ms
----------------------------------------Completed Test----------------------------------------
```

**With this **PRs** Build** :
```python3
python3 test_cugraph_sampling.py


Sampling 1,000 took = 73.62375259399414 ms
Sampling 10,000 took = 87.1060848236084 ms
Sampling 100,000 took = 134.7532033920288 ms
----------------------------------------Completed Test----------------------------------------
```

### CODE REVIEW

File: benchmarks/shared/build_cugraph_ucx/build-ucx.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+# Copyright (c) 2022, NVIDIA CORPORATION.
```
Comment (alexbarghi-nv) [inline benchmarks/shared/build_cugraph_ucx/build-ucx.sh:2]:
```suggestion
# Copyright (c) 2023, NVIDIA CORPORATION.
```

File: benchmarks/shared/build_cugraph_ucx/cugraph_ucx.dockerfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2018-2022, NVIDIA CORPORATION.
```
Comment (alexbarghi-nv) [inline benchmarks/shared/build_cugraph_ucx/cugraph_ucx.dockerfile:1]:
```suggestion
# Copyright (c) 2022-2023, NVIDIA CORPORATION.
```

File: benchmarks/shared/build_cugraph_ucx/test_client_bandwidth.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2018-2022, NVIDIA CORPORATION.
```
Comment (alexbarghi-nv) [inline benchmarks/shared/build_cugraph_ucx/test_client_bandwidth.py:1]:
```suggestion
# Copyright (c) 2022-2023, NVIDIA CORPORATION.
```

File: benchmarks/shared/build_cugraph_ucx/test_cugraph_sampling.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2018-2022, NVIDIA CORPORATION.
```
Comment (alexbarghi-nv) [inline benchmarks/shared/build_cugraph_ucx/test_cugraph_sampling.py:1]:
```suggestion
# Copyright (c) 2022-2023, NVIDIA CORPORATION.
```

**Review by alexbarghi-nv — CHANGES_REQUESTED** [review]
needs copyright fix

**Review by alexbarghi-nv — APPROVED** [review]

File: benchmarks/shared/build_cugraph_ucx/cugraph_ucx.dockerfile [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    python=$PYTHON_VER \
+    setuptools \
+    tqdm
+
+
+# Build ucx from source with IB support 
+# on 1.14.x
+RUN conda remove --force -y ucx ucx-proc
+
+ADD build-ucx.sh /home/build-ucx.sh
```
[the remark this answers is not in the document]
Comment (VibhuJawa) [reply benchmarks/shared/build_cugraph_ucx/cugraph_ucx.dockerfile:62]: No reason.  Just an arbitrary choice,  I just ended up using /home, can switch to any thing better you suggest.

File: benchmarks/shared/build_cugraph_ucx/test_client_bandwidth.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import rmm
+from time import time
+
+def benchmark_func(func, n_times=10):
+    def wrap_func(*args, **kwargs):
+        time_ls = []
+        # ignore 1st run
+        # and return other runs
+        for _ in range(0,n_times+1):
+            t1 = time()
```
[the remark this answers is not in the document]
Comment (VibhuJawa) [reply benchmarks/shared/build_cugraph_ucx/test_client_bandwidth.py:29]: Let me switch this to `perf_counter` .

File: benchmarks/shared/build_cugraph_ucx/cugraph_ucx.dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (rlratzel) [reply benchmarks/shared/build_cugraph_ucx/cugraph_ucx.dockerfile:62]: I pulled and ran the image and the default user is `root`, and `HOME` is `/root`, so maybe `/root`?  I don't have a strong preference though.

[the remark this answers is not in the document]
Comment (VibhuJawa) [reply benchmarks/shared/build_cugraph_ucx/cugraph_ucx.dockerfile]: Resolved by https://github.com/rapidsai/cugraph/pull/3088/commits/88a8865b5b20c615bbff64c04e8e1f202a24e287 .

File: benchmarks/shared/build_cugraph_ucx/test_client_bandwidth.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (VibhuJawa) [reply benchmarks/shared/build_cugraph_ucx/test_client_bandwidth.py]: Resolved by https://github.com/rapidsai/cugraph/pull/3088/commits/88a8865b5b20c615bbff64c04e8e1f202a24e287  .

**Comment by VibhuJawa** [comment]
@rlratzel , Addressed your reviews, please merge if this is good to merge.

**Comment by rlratzel** [comment]
@gpucibot merge

**Pull request merged**